SwanCustomEnvironments/swancustomenvironments/templates/customenvs.html

```diff
@@ -0,0 +1,157 @@
+
+{% block stylesheet %}
+{{super()}}
+<style type="text/css">
+/* disable initial hide */
+div#header, div#site {
+    display: block;
+}
+#output-box {
+	display: block;
+	width: 1150px;
+	height: 300px;
+	overflow-y: scroll;
+	overflow-x: hidden;
+	white-space: pre-wrap;
+	text-align: left;
+	border: 1px solid #ccc;
+	background-color: #f7f7f7;
+	padding: 10px;
+	border-radius: 5px;
+}
+#back_button {
+	display: none;
+	text-align: center;
+    background-color: #fc6404;
+	position: absolute;
+	left: 50%;
+	transform: translateX(-50%);
+    color: white;
+    border: none;
+    border-radius: 5px;
+    padding: 10px 20px;
+    font-size: 16px;
+    cursor: pointer;
+    transition: background-color 0.3s ease;
+}
+#next_button {
+	display: none;
+	text-align: center;
+    background-color: #0454a4;
+	position: absolute;
+	left: 50%;
+	transform: translateX(-50%);
+    color: white;
+    border: none;
+    border-radius: 5px;
+    padding: 10px 20px;
+    font-size: 16px;
+    cursor: pointer;
+    transition: background-color 0.3s ease;
+}
+</style>
+{% endblock %}
+
+{% block params %}
+{{super()}}
+data-base-url="{{base_url | urlencode}}"
+{% endblock %}
+
+{% block site %}
+
+<div class="container">
+	<div class="row">
+		<div class="swan-info">
+			<div id="swan-loader">
+				<div class="loader-circle">
+					<img src="{{hub_prefix}}static/swan/logos/{{swan_logo_filename}}">
+				</div>
+				<div class="loader-line-mask">
+					<div class="loader-line"></div>
+				</div>
+				<span class="text">Creating environment...</span>
+			</div>
+		</div>
+		<br><br>
+		<div id="output-box"></div>
+		<br>
+		<button id="back_button">Back</button>
+		<button id="next_button">Next</button>
+	</div>
+</div>
+
+{% endblock %}
+
+{% block script %}
+    {{super()}}
+
+<script type = "text/javascript" >
+	
+	function error(reason) {
+		var loader = document.getElementById('swan-loader');
+		loader.getElementsByClassName('loader-line-mask')[0].style.display = 'none';
+		loader.getElementsByClassName('text')[0].innerHTML = 'Error creating environment';
+	}
+
+	var base_url = document.body.getAttribute('data-base-url');
+	
+	const urlParams = new URLSearchParams(window.location.search);
+	var decoded_req = urlParams.get('req');
+	var env_name = encodeURIComponent(urlParams.get('env') || "");
+	var requirements = encodeURIComponent(decoded_req || "");
+	var clear = encodeURIComponent(urlParams.get('clear') || "true");
+	var accpy_version = encodeURIComponent(urlParams.get('accpy') || "");
+	var custom_python = encodeURIComponent(urlParams.get('python') || "");
+	
+	var project_folder = '/';
+	var path_folders = urlParams.get('req').split('/');
+	if (requirements.startsWith('http')) {
+		// Extract repository name from URL
+		const pattern = /^https?:\/\/[^/]+\/([^/\s]+)\/([^/\s]+)/;
+		const match = urlParams.get('req').match(pattern);
+		project_folder = match ? `/tree/SWAN_projects/${match[2]}` : '';
+	} else if (path_folders.length > 6) {
```

Comment:
Why greater than 6?

Reply:
/eos/user/u/username/folder1/folder2/folder3/requirements.txt

we gotta get the folder1/folder2/folder3 path to inject into url path. how, requirements.txt can just be within /eos/user/u/username, which means the url path cannot have that /tree in the url

Reply:
Sorry I still don't get it, why greater than `6`? Why is the `/eos/user/u/username/folder1/folder2/requirements.txt` case different than `/eos/user/u/username/folder1/folder2/folder3/requirements.txt`? If I understand correctly you want to match `/eos/user/u/username`, strip out that group, and generate `/tree/folder1/...` ?

Reply:
those two cases aint differente cause both got more that 6 elements. but yeah, thats the idea. i just want to gather the middle subfolders to inject into the url, and prevent cases like `/eos/user/u/username/requirements.txt` where the file is in the eos home directory. in this case you dont even need to change your url. i'll write a comment to breefly explain that

Reply:
Ok, so why is `/eos/user/u/username/folder1/requirements.txt` different? That should transform into `tree/folder1/requirements.txt`, right?

And in case of `/eos/user/u/username/requirements.txt`, you can't transform into `tree/requirements.txt`?

It's just that the condition seems a bit cryptic (the >6). Usually these things are done via pattern matching. If you just want to get rid of the `/eos/user/u/username` part of the URL, you match that with a group in a pattern (see `re` package in Python) and strip it out.

Reply:
Yes, you're right. I definitely overcomplicated things unnecessarily